atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/HostIdEvolution.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * (c) Copyright 2023 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.keyvalue.cassandra;
+
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Sets;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public final class HostIdEvolution {
+    private HostIdEvolution() {
+        // Utility class
+    }
+
+    /**
+     * Returns true iff there exists a plausible sequence of cluster changes, measured through differences in snapshots
```

Comment:
Let's define the plausible evolution, since there is a possible interpretation that the empty set is the plausible evolution from the empty set, which is not what's wanted here.